modal-go/sandbox.go

```diff
@@ -606,8 +693,46 @@ func (sb *Sandbox) ensureTaskID(ctx context.Context) error {
 	return nil
 }
 
+func (sb *Sandbox) getOrCreateCommandRouterClient(ctx context.Context, taskID string) (*TaskCommandRouterClient, error) {
+	sb.mu.Lock()
+	defer sb.mu.Unlock()
+
+	if sb.commandRouterClient == nil {
+		client, err := TryInitTaskCommandRouterClient(
+			ctx,
+			sb.client.cpClient,
+			taskID,
+			sb.client.logger,
+			sb.client.profile,
+		)
+		if err != nil {
+			return nil, err
+		}
+		if client == nil {
+			return nil, fmt.Errorf("command router access is not available for this sandbox")
+		}
+		sb.commandRouterClient = client
+	}
+	return sb.commandRouterClient, nil
+}
+
+// Close closes any resources associated with the Sandbox.
+// This should be called when the Sandbox is no longer needed
+// in the local client. The sandbox can still be running and
+// accessed in other clients.
+func (sb *Sandbox) Close() {
+	sb.mu.Lock()
+	defer sb.mu.Unlock()
+
+	if sb.commandRouterClient != nil {
+		_ = sb.commandRouterClient.Close()
```

Comment:
Does calling `commandRouterClient.Close` also end all the goroutines spawned by `ExecStdioRead`?

Reply:
Pushed a change in 3769bc8 now that propagates cancellation.